src/core/main/worker/worker_main.ts

```diff
@@ -47,6 +47,8 @@ import {
 } from "./globals";
 import sendMessage, { formatErrorForSender } from "./send_message";
 
+let shouldMonitorSegmentStoreUpdates = false;
```

Comment:
I'm not really satisfied about the place this flag is declared, but I'm not sure where to put it while keeping same behavior

Reply:
What about another solution where it's the main thread who always request it (and decides of when to do it in some sort of interval)?

Kind of like a
```
-> ping
<- pong
```
system?

Though it also has its own drawbacks